File: Informatik/Py/getPrimeFactors.py
import math

num = int(input('Enter a number to get its Prime Factors: '))
factors = getPrimeFactors(num)
print(f"Prime factors of {num} are: {factors}")

def getPrimeList(n):
    primes = []
    for num in range(2, int(math.isqrt(n)) + 1):
        is_prime = True
        for p in primes:
            if num % p == 0:
                is_prime = False
                break
        if is_prime:
            primes.append(num)
    return primes

def getPrimeFactors(num):
    primeList = getPrimeList(num)
    numPrimeFactors = []
    primeIndex = 0

    while num > 1 and primeIndex < len(primeList):
        currentPrime = primeList[primeIndex]
        if num % currentPrime == 0:
            numPrimeFactors.append(currentPrime)
            num //= currentPrime
        else:
            primeIndex += 1

    if num > 1:
        numPrimeFactors.append(num)

    return numPrimeFactors

File: Informatik/Py/getPrimeFactors.php
<?php

function primeFactors($num): array {

    $primeList = [];
    $limit = floor(sqrt($num));
    for ($i = 2; $i <= $limit; $i++) {
        $isPrime = true;
        for ($j = 2; $j <= sqrt($i); $j++) {
            if ($i % $j == 0) {
                $isPrime = false;
                break;
            }
        }
        if ($isPrime) {
            $primeList[] = $i;
        }
    }

    $primeNumberIndex = 0;

    while ($num > 1 && $primeNumberIndex < count($primeList)) {
        $currentPrime = $primeList[$primeNumberIndex];
        if ($num % $currentPrime == 0) {
            $numPrimeFactors[] = $currentPrime;
            $num = $num / $currentPrime;
        } else {
            $primeNumberIndex++;
        }
    }

    if ($num > 1) {
        $numPrimeFactors[] = $num;
    }

    return $numPrimeFactors;
}

$num = 84;
$factors = primeFactors($num);
echo "Primfaktoren von $num: " . implode(", ", $factors);
